packages/grid/_modules_/grid/hooks/features/sorting/useGridSorting.ts

```diff
@@ -299,11 +298,6 @@ export const useGridSorting = (
   };
   useGridApiMethod(apiRef, sortApi, 'GridSortApi');
 
-  React.useEffect(() => {
-    // When the rows prop change, we re apply the sorting.
-    apiRef.current.applySorting();
-  }, [apiRef, props.rows]);
-
   React.useEffect(() => {
     if (rowCount > 0) {
       logger.debug('row changed, applying sortModel');
```

Comment:
I wonder if we can get ride of this effect too

```suggestion
```

Reply:
Why were we reapplying the sort when the `rowCount` selector changes AND when `props.rows` changes ?
For me we can remove it.

All the test are passing.

With this last `useEffect` removal, `useGridFilter` and `useGridSorting` both have 2 `useEffect` :

- the control state registration (that will become synchronous with the sync state initialization)
- the prop to state (that could become synchronous but with some issues that I must address 1st)

It would be cool to have two major pieces of our architecture that have 100% or their state updates synchronous.
We may not succeed in having all our state updates synchronous (some edge cases are hard to solve).
But if when the user updates its rows, our internal state have directly the new sorted and filtered rows ready it would be cool (that needs some work after this PR of course).

Reply:
I added a commit removing the `useEffect` you linked.